web/client/components/import/dragZone/enhancers/processFiles.jsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const Rx = require('rxjs');
+const { compose, mapPropsStream, createEventHandler} = require('recompose');
+const FileUtils = require('../../../../utils/FileUtils');
+const LayersUtils = require('../../../../utils/LayersUtils');
+
+const JSZip = require('jszip');
+
+const tryUnzip = (file) => {
+    return FileUtils.readZip(file).then((buffer) => {
+        var zip = new JSZip();
+        return zip.loadAsync(buffer);
+    });
+};
+
+/**
+ * Checks if the file is allowed. Returns a promise that does this check.
+ */
+const checkFileType = (file) => {
```

Comment:
Are we duplicating the existing Shapefile code, instead of putting it in a shared place?

Reply:
Yes I know, but I didn't have time to refactor shapefile plugin that has an ugly implementation and tests that doesn't make you sure you did all things correctly. In addition, I think that the shapfile plugin could be removed in the future , replaced by this (more functionalities, better UI, so I didn't wanted to spend time refactoring a component that may be removed.